duo_openvpn_as.py

```diff
@@ -204,20 +203,36 @@ def sign(ikey, skey, method, host, uri, date, sig_version, params):
     Return basic authorization header line with a Duo Web API signature.
     """
     canonical = canonicalize(method, host, uri, params, date, sig_version)
-    if isinstance(skey, unicode):
+    if not isinstance(skey, str):
         skey = skey.encode('utf-8')
-    sig = hmac.new(skey, canonical, hashlib.sha1)
+
+    try:
+        sig = hmac.new(skey, canonical, hashlib.sha1)
+
+    except TypeError:
+        # python3 require bytes
+        skey = skey.encode('utf-8')
+        sig = hmac.new(skey, canonical.encode(), hashlib.sha1)
+
     auth = '%s:%s' % (ikey, sig.hexdigest())
-    return 'Basic %s' % base64.b64encode(auth)
+
+    try:
```

Comment:
what you have here seems like it should work, but for consistency I'd prefer if we could mimic what we do in duo_client_python
```
auth = '%s:%s' % (ikey, sig.hexdigest())
if isinstance(auth, text_type):
        auth = auth.encode('utf-8')
b64 = base64.b64encode(auth)
if not isinstance(b64, six.text_type):
    b64 = b64.decode('utf-8')
return 'Basic %s' % b64
```

Reply:
All the unicode stuff are changed to the six text_type recommendation.